app/src/main/java/com/infomaniak/drive/ui/menu/GalleryViewModel.kt

```diff
@@ -36,28 +36,46 @@ class GalleryViewModel : ViewModel() {
     private var currentCursor: String? = null
 
     val galleryApiResult = MutableLiveData<Pair<ArrayList<File>, IsComplete>?>()
+    private var lastGalleryFiles = ArrayList<File>()
+
+    val needToRestoreFiles get() = galleryApiResult.isInitialized
 
     fun loadLastGallery(driveId: Int, ignoreCloud: Boolean) {
-        getGalleryJob?.cancel()
-        getGalleryJob = viewModelScope.launch(Dispatchers.IO) {
-            galleryApiResult.postValue(getLastGallery(driveId, isFirstPage = true, ignoreCloud))
-        }
+        lastGalleryFiles = arrayListOf()
+        loadLastGallery(driveId, ignoreCloud, isFirstPage = true)
     }
 
     fun loadMoreGallery(driveId: Int, ignoreCloud: Boolean) {
         currentCursor?.let {
-            getGalleryJob?.cancel()
-            getGalleryJob = viewModelScope.launch(Dispatchers.IO) {
-                galleryApiResult.postValue(getLastGallery(driveId, isFirstPage = false, ignoreCloud, currentCursor))
-            }
+            loadLastGallery(driveId, ignoreCloud, isFirstPage = false, currentCursor)
         }
     }
 
-    private fun getLastGallery(
+    fun restoreGalleryFiles() {
+        if (needToRestoreFiles) {
+            galleryApiResult.value = lastGalleryFiles to galleryApiResult.value!!.second
```

Comment:
Is it something we really want to crash here ? In my opinion, It would be better to display an error with an empty state instead of crashing.